src/utill.rs

```diff
@@ -238,6 +240,26 @@ mod tests {
         str_to_bitcoin_network(net_str);
     }
 
+    #[tokio::test]
+    async fn test_send_message() {
+        let listener = TcpListener::bind("127.0.0.1:0").await.unwrap();
+        let address = listener.local_addr().unwrap();
+
+        tokio::spawn(async move {
+            let (mut socket, _) = listener.accept().await.unwrap();
+            let mut buffer = vec![0u8; 1024];
```

Comment:
This would also work with `Vec::new()`. So you don't have to pre-guess the buffer length.

Reply:
Ok. On it